Simbody/include/simbody/internal/SmoothSphereHalfplaneForce.h

```diff
@@ -0,0 +1,126 @@
+#ifndef SimTK_SIMBODY_SMOOTH_SPHERE_HALFPLANE_FORCE_H_
+#define SimTK_SIMBODY_SMOOTH_SPHERE_HALFPLANE_FORCE_H_
+
+/* -------------------------------------------------------------------------- *
+ *                               Simbody(tm)                                  *
+ * -------------------------------------------------------------------------- *
+ * This is part of the SimTK biosimulation toolkit originating from           *
+ * Simbios, the NIH National Center for Physics-Based Simulation of           *
+ * Biological Structures at Stanford, funded under the NIH Roadmap for        *
+ * Medical Research, grant U54 GM072970. See https://simtk.org/home/simbody.  *
+ *                                                                            *
+ * Portions copyright (c) 2008-19 Stanford University and the Authors.        *
+ * Authors: Antoine Falisse, Gil Serrancoli                                   *
+ * Contributors: Peter Eastman                                                *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include "SimTKcommon.h"
+
+#include "simbody/internal/common.h"
+#include "simbody/internal/Force.h"
+
+namespace SimTK {
+
+class SmoothSphereHalfplaneForceImpl;
+
+/**
+ * This class models the forces generated by simple point contacts between a
+ * sphere and a half space. The contact model is a smooth (i.e., twice
+ * continuously differentiable) approximation of the HuntCrossleyForce already
+ * available in Simbody. The proposed implementation was designed for use
+ * with gradient-based optimization algorithms and algorithmic/automatic
+ * differentiation. To this aim, conditional if statements were approximated by
+ * using hyperbolic tangent functions. For example, the following if statement:
+ * <pre>     y = 0, if x < d </pre>
+ * <pre>     y = a, if x >= d </pre>
+ * can be approximated by:
+ * <pre>     f = 0.5 + 0.5 tanh(b(x-d)) </pre>
+ * <pre>     y = a f </pre>
+ * where b is a parameter determining the smoothness of the transition.
```

Comment:
This class also applies force when the sphere is not in contact, correct?

It would be great to add a plot to the doxygen showing the force as a function of depth. I could help with this.